modules/mod_translation/controllers/controller_language_set.erl

```diff
@@ -51,10 +51,10 @@ moved_temporarily(ReqData, Context0) ->
     Context = ?WM_REQ(ReqData, Context0),
     Context1 = mod_translation:set_user_language(z_context:get_q("code", Context), Context),
     Page = z_context:get_q("p", Context1),
-    Location = case z_utils:is_empty(Page) of
-                   true -> "/";
-                   false -> Page
-               end,
+    Location = case Page of
+        "/" ++ _ -> Page;
+        _ -> "/"
+    end,
     AbsUrl = z_context:abs_url(add_language(mod_translation:url_strip_language(Location), Context1), Context1),
```

Comment:
Shouldn't Location also be fed through a `z_context:site_url`? As to have it sanitized?

Reply:
Might be better, indeed.  Or at least pass Location through the URI sanitizer.

Reply:
Added the sanitize.